### PR TITLE
Cleanup WebglProgram

### PR DESCRIPTION
**Pull request by gero3**
This seperates more pieces of code. 

 - ShadowTypeDefine gets defined in its [own function](https://github.com/mrdoob/three.js/blob/75db8bf7b50297efdc57f4efc934d017b3b66f02/src/renderers/webgl/WebGLProgram.js#L268-L288) instead of in [the WEBGLProgram Constructor](https://github.com/mrdoob/three.js/blob/f320bfc05e735f1e4f7e0a08327b60e0387a0091/src/renderers/webgl/WebGLProgram.js#L255-L269)
- envMapTypeDefine gets defined in its [own function](https://github.com/mrdoob/three.js/blob/75db8bf7b50297efdc57f4efc934d017b3b66f02/src/renderers/webgl/WebGLProgram.js#L290-L323) instead of in [the WEBGLProgram Constructor](https://github.com/mrdoob/three.js/blob/f320bfc05e735f1e4f7e0a08327b60e0387a0091/src/renderers/webgl/WebGLProgram.js#L271-L325)
- envMapModeDefine gets defined in its [own function](https://github.com/mrdoob/three.js/blob/75db8bf7b50297efdc57f4efc934d017b3b66f02/src/renderers/webgl/WebGLProgram.js#L325-L344) instead of in [the WEBGLProgram Constructor](https://github.com/mrdoob/three.js/blob/f320bfc05e735f1e4f7e0a08327b60e0387a0091/src/renderers/webgl/WebGLProgram.js#L271-L325)
- envMapBlendingDefine gets defined in its [own function](https://github.com/mrdoob/three.js/blob/75db8bf7b50297efdc57f4efc934d017b3b66f02/src/renderers/webgl/WebGLProgram.js#L346-L372) instead of in [the WEBGLProgram Constructor](https://github.com/mrdoob/three.js/blob/f320bfc05e735f1e4f7e0a08327b60e0387a0091/src/renderers/webgl/WebGLProgram.js#L271-L325)

### CODE REVIEW

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
Looking good!

File: src/renderers/webgl/WebGLProgram.js [hunk 1]
```diff
@@ -243,14 +243,29 @@ function unrollLoops( string ) {
 
 }
 
-function WebGLProgram( renderer, extensions, code, material, shader, parameters, capabilities ) {
+function generatePrecision( parameters ) {
 
-	var gl = renderer.getContext();
+	var precisionstring = `precision ${parameters.precision} float;\nprecision ${parameters.precision} int;`;
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLProgram.js]: Do you mind doing the template literal change in another PR?

**Comment by gero3** [comment]
The string interpolation is removed now.

**Comment by mrdoob** [comment]
Screenshot 2019-09-15 at 11 50 00

Oh? I've never seen that before 🤔

**Comment by gero3** [comment]
I'll redo this one. I don't know what happened.

**Pull request closed without merging**